### PR TITLE
Line Graph Integration

### PR DESCRIPTION
**Pull request by nxjiten**
@natabal 
Integrate line graph using **recharts** library. Please have a look.

### CODE REVIEW

File: GraphsV2/LineGraph/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        data,
+        key: linesColumn,
+        xColumn,
+        yColumn,
+    });
+
+    const [tooltipName, setToolTipKey] = useState(-1);
+    sortByKey(parsedData, 'ts');
+    
+    const legendHeight = (legend.separate * height) / 100;
```
Comment (natabal) [inline GraphsV2/LineGraph/index.js]: please move this calculation to renderLEgend

File: GraphsV2/LineGraph/index.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    } = properties;
+
+    const { parsedData, uniqueKeys: lineKeys } = dataParser({
+        data,
+        key: linesColumn,
+        xColumn,
+        yColumn,
+    });
+
+    const [tooltipName, setToolTipKey] = useState(-1);
```
Comment (natabal) [inline GraphsV2/LineGraph/index.js]: this should be tooltipKey not tooltipName

File: GraphsV2/LineGraph/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                            onMouseLeave={() => setToolTipKey(-1)}
+                            type="linear" name={lineItem} dataKey={lineItem} stroke={color} fill={color} />
+                    )
+                })
+            }
+        </LineChart>
+    );
+}
+
+const sortByKey = (array, key) => {
```
Comment (natabal) [inline GraphsV2/LineGraph/index.js]: this function should be called sortAscending and please move it to the util it's not specific to this graph, even better parsedData should send you a sorted data if required.

File: GraphsV2/LineGraph/index.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                })
+            }
+            {
+                customTooltip({ tooltip, tooltipName })
+            }
+            {
+                lineKeys.map((lineItem, index) => {
+                    const color = colors[index % 20];
+                    return (
+                        <Line connectNulls={zeroStart}
```
Comment (natabal) [inline GraphsV2/LineGraph/index.js]: please provide a 'key' to Line and an id as well.

File: GraphsV2/LineGraph/index.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                    legend,
+                    legendHeight,
+                })
+            }
+            {
+                customTooltip({ tooltip, tooltipName })
+            }
+            {
+                lineKeys.map((lineItem, index) => {
+                    const color = colors[index % 20];
```
Comment (natabal) [inline GraphsV2/LineGraph/index.js]: again % 20 without an explanation. Why not have a common function to choose the color

File: GraphsV2/LineGraph/index.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                    return (
+                        <Line connectNulls={zeroStart}
+                            onMouseEnter={({ name }) => setToolTipKey(name)}
+                            onMouseLeave={() => setToolTipKey(-1)}
+                            type="linear" name={lineItem} dataKey={lineItem} stroke={color} fill={color} />
```
Comment (natabal) [inline GraphsV2/LineGraph/index.js]: split props one per line.

**Comment by nxjiten** [comment]
chrome-capture
rsz_screenshot_from_2020-02-11_09-28-33
@natabal 
Please find the attached screenshot.

**Pull request merged**